test/e2e/validation_test.go

```diff
@@ -143,3 +150,261 @@ func testNodeTaint(t *testing.T) {
+		return errors.Wrap(err, "error ensuring RoleBinding created")
+	}
+	return nil
+}
+
+// copyKeySecret copies the current private-key secret from the WMCO namespace into the testContext's test namespace
+func (tc *testContext) copyKeySecret() error {
+	currentSecret, err := tc.client.K8s.CoreV1().Secrets(tc.namespace).Get(context.TODO(), secrets.PrivateKeySecret, meta.GetOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to get current private key secret from WMCO namespace")
+	}
+
+	// Remove secret from test namespace if it already exists, so the new secret can be created
+	secretClient := tc.client.K8s.CoreV1().Secrets(tc.workloadNamespace)
+	err = secretClient.Delete(context.TODO(), secrets.PrivateKeySecret, meta.DeleteOptions{})
+	if err != nil && !apierrors.IsNotFound(err) {
+		return errors.Wrap(err, "unable to delete existing copied secret")
+	}
+	newSecret := core.Secret{
+		ObjectMeta: meta.ObjectMeta{Name: secrets.PrivateKeySecret},
+		Data:       currentSecret.Data,
+	}
+	_, err = secretClient.Create(context.TODO(), &newSecret, meta.CreateOptions{})
+	if err != nil {
+		return errors.Wrap(err, "could not create secret")
+	}
+	return nil
+}
+
+// getOpenShiftToolsImage returns a pullable image from the openshift/tools imagestream
+func (tc *testContext) getOpenShiftToolsImage() (string, error) {
+	imageStream, err := tc.client.Images.ImageStreams("openshift").Get(context.TODO(), "tools", meta.GetOptions{})
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get openshift/tools imagestream")
+	}
+	image, _, _, _, err := imageutil.ResolveRecentPullSpecForTag(imageStream, "latest", false)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get latest debug image from imagestream")
+	}
+	return image, nil
+}
+
+// runSSHJob creates and waits for a Kubernetes job to run. The command provided will be executed on the host specified
+// by the provided IP.
+func (tc *testContext) runSSHJob(name, command, ip string) (string, error) {
+	if err := tc.sshSetup(); err != nil {
+		return "", errors.Wrap(err, "unable to fufill SSH requirements")
+	}
+
+	// Copy the current WMCO private key to the test namespace, so that the SSH pod which will run in the test
+	// namespace will have access to it
+	if err := tc.copyKeySecret(); err != nil {
+		return "", errors.Wrap(err, "error copying private-key secret to test namespace")
+	}
+
+	// The SSH pod will use the openshift/tools image, the same image used by `oc debug`. This image is available on
+	// all OpenshiftClusters, and has SSH pre-installed.
+	debugImage, err := tc.getOpenShiftToolsImage()
+	if err != nil {
+		return "", errors.Wrap(err, "error getting debug image")
+	}
+
+	// username will be Administrator on all cloud providers, except Azure where it is "capi"
+	var username string
+	if tc.CloudProvider.GetType() == config.AzurePlatformType {
+		username = "capi"
+	} else {
+		username = "Administrator"
+	}
```

Comment:
You should do this once and pass in the user into this function.

Reply:
Removing the duplicate code doesn't seem possible without making the username result part of the testContext struct. I've made that change for now, but I'm not sure if thats what you wanted. I can also make this a helper function like @saifshaikh48 suggested.

The username needs to be known in the creation suite when creating the windows-instance configmap and when SSHing in to validate creation, as well as in the deletion suite in the future when SSHing in to validate proper deconfiguration.